build-system/tasks/release/index.js

```diff
@@ -0,0 +1,375 @@
+  }
+}
+
+async function fetchAmpSw_(distFlavorTypes, outputTempDir) {
+  for (const flavorType of distFlavorTypes) {
+    await fs.ensureDir(path.join(outputTempDir, flavorType, 'dist/sw'));
+  }
+
+  const ampSwBaseTempDir = path.join(outputTempDir, 'base/dist/sw');
+
+  const ampSwTarballUrl = await fetch(
+    'https://registry.npmjs.org/@ampproject/amp-sw'
+  )
+    .then((res) => res.json())
+    .then((json) => {
+      const {latest} = json['dist-tags'];
+      return json.versions[latest].dist.tarball;
+    });
+  await fetch(ampSwTarballUrl).then(
+    (res) =>
+      new Promise((resolve) => {
+        const tarWritableStream = tar.extract({
+          cwd: ampSwBaseTempDir,
+          filter: (path) => path.startsWith('package/dist'),
+          strip: 2, // to strip "package/dist/".
+        });
+        res.body.pipe(tarWritableStream);
+        tarWritableStream.on('end', resolve);
+      })
+  );
+
+  await Promise.all(
+    distFlavorTypes
+      .filter((flavorType) => flavorType != 'base')
+      .map((flavorType) =>
+        fs.copy(
+          ampSwBaseTempDir,
+          path.join(outputTempDir, flavorType, 'dist/sw')
+        )
+      )
+  );
+
+  logSeparator_();
+}
+
+async function populateOrgCdn_(distFlavors, outputTempDir, outputDir) {
+  for (const {flavorType, rtvPrefixes} of distFlavors) {
+    await Promise.all(
+      rtvPrefixes.map((rtvPrefix) => {
+        const rtvNumber = `${rtvPrefix}${VERSION}`;
+        const rtvPath = path.join(outputDir, 'org-cdn/rtv', rtvNumber);
+        return fs
+          .ensureDir(rtvPath)
+          .then(() =>
+            fs.copy(path.join(outputTempDir, flavorType, 'dist'), rtvPath)
+          );
+      })
+    );
+  }
+
+  logSeparator_();
+}
+
+async function prependConfig_(distFlavors, outputDir) {
+  const activeChannels = Object.entries(CHANNEL_CONFIGS).filter(
+    ([rtvPrefix]) => {
+      const rtvNumber = `${rtvPrefix}${VERSION}`;
+      const rtvPath = path.join(outputDir, 'org-cdn', rtvNumber);
+      return fs.pathExistsSync(path.join(rtvPath));
+    }
+  );
+
+  const allPrependPromises = [];
+  for (const [rtvPrefix, channelConfig] of activeChannels) {
+    const rtvNumber = `${rtvPrefix}${VERSION}`;
+    const rtvPath = path.join(outputDir, 'org-cdn/rtv', rtvNumber);
+    const channelPartialConfig = {
+      v: rtvNumber,
+      type: channelConfig.type,
+      ...require(`../../global-configs/${channelConfig.configBase}-config.json`),
+    };
+
+    allPrependPromises.push(
+      ...TARGETS_TO_CONFIG.map((target) => {
+        const targetPath = path.join(rtvPath, target.file);
+        const channelConfig = JSON.stringify({
+          ...channelPartialConfig,
+          ...target.config,
+        });
+
+        fs.readFile(targetPath, 'utf-8').then((contents) =>
+          fs.writeFile(
+            targetPath,
+            `self.AMP_CONFIG=${channelConfig};/*AMP_CONFIG*/${contents}`
+          )
+        );
+      })
+    );
+  }
+
+  logSeparator_();
+}
```

Comment:
I think this function can be simplified by using `applyConfig()`:

https://github.com/ampproject/amphtml/blob/6c6c63c972d0d37c791ee492398f195503a8e02c/build-system/tasks/prepend-global/index.js#L121-L131

(Or punt if this requires non-trivial changes to `prepend-global/index.js`.)

Reply:
I took a look at it right now and it would require a relatively trivial change to `applyConfig`, but I prefer to leave it as a TODO for now. Note that there is a difference in the output!

```js
// `applyConfig` prepends this:
`self.AMP_CONFIG||(self.AMP_CONFIG=${configString});/*AMP_CONFIG*/`

// Google's release script prepends this:
`self.AMP_CONFIG=${channelConfig};/*AMP_CONFIG*/`
```

Do you have any context as to why there's a difference?

Reply:
Good question. Not sure. Punting to a separate PR sounds like the prudent thing to do.

Reply:
Based on offline conversations, it appears `prepend-global` tries to account for previously added configs during development, while the release scripts don't need to do it. Context: #3924.

I still think it's worth doing a small refactor to `applyConfig` to retain this difference and re-use it in `gulp release`. (Separate PR works if you prefer.)

Resolving this comment.